ContributedExamples/surface_water-mass_transformation.ipynb

```diff
@@ -0,0 +1,782 @@
+    "    ds = xr.Dataset({'surface_water_mass_transformation': net_transformation, 'time_bounds': time_bounds})\n",
+    "    ds.to_netcdf(outpath+'/SWMT_'+time_bounds+'.nc')\n",
+    "    del(net_transformation)\n",
+    "    return outpath, time_bounds ## helpful for re-loading"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "salt_transformation binning done\n",
+      "heat_transformation binning done\n",
+      "CPU times: user 2min 33s, sys: 2min 4s, total: 4min 37s\n",
+      "Wall time: 5min 16s\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%time\n",
+    "outpath = '/g/data/v45/rm2389/SWMT/01deg_jra55v13_ryf9091' # use a place where you have write permission\n",
+    "outpath, time_bounds = save_SWMT(expt, session, '1904-01', '1905-12', outpath)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 4. Reload computed surface water-mass transformation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_SWMT(outpath, time_bounds):\n",
+    "    net_transformation = xr.open_dataset(outpath+'/SWMT_'+time_bounds+'.nc',chunks={'isopycnal_bins':1})\n",
+    "    net_transformation = net_transformation.surface_water_mass_transformation\n",
+    "    heat_transformation = xr.open_dataset(outpath+'/HT_'+time_bounds+'.nc',chunks={'isopycnal_bins':1})\n",
+    "    heat_transformation = heat_transformation.binned_heat_transformation\n",
+    "    salt_transformation = xr.open_dataset(outpath+'/ST_'+time_bounds+'.nc',chunks={'isopycnal_bins':1})\n",
+    "    salt_transformation = salt_transformation.binned_salt_transformation\n",
+    "    return  net_transformation, heat_transformation, salt_transformation"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 5. Plotting options: summed over southern ocean, south of 59S"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "One useful way of visualising the surface water-mass transformation metric is to sum the transformation rates (in density bins) over the analysis region, which here is the entire southern ocean south of 59S."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "area_t = cc.querying.getvar(expt,'area_t',session, n=1) # needed for the plots I'll provide.\n",
+    "swmt, heat, salt = get_SWMT(outpath, time_bounds)\n",
+    "## sum over region, convert to Sv, -1 is so that positive swmt values indicate transformation into denser classes\n",
+    "## (you can have it either way)\n",
+    "isopycnal_bin_mid = swmt.isopycnal_bins\n",
+    "swmt_sum = (swmt * area_t/1e6).sum(dim = 'xt_ocean').sum(dim = 'yt_ocean').values\n",
```

Comment:
I believe you can sum across multiple dimensions simultaneously. It may or may not be more efficient (haven't actually benchmarked that...)